app/models/task.py

```diff
@@ -1,5 +1,60 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from sqlalchemy import ForeignKey
+from datetime import datetime
+from sqlalchemy import DateTime, Boolean
+from typing import Optional
+from ..models.goal import Goal
+
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]]
+    is_complete: Mapped[bool] = mapped_column(Boolean, default=False)
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+    def to_dict(self):
+        task_as_dict = {}
+        task_as_dict["id"] = self.id
+        task_as_dict["title"] = self.title
+        task_as_dict["description"] = self.description
+        task_as_dict["completed_at"] = self.completed_at
+        task_as_dict["is_complete"] = self.is_complete
```

Comment:
`self` should not have an attribute `is_complete`. It should be calculated from `completed_at`.

What would the logic look like to use `Task`s attribute `completed_at` to set `is_complete` to either True or False? Perhaps, you could look at `completed_at` and if it is None then `is_complete` would be set to False and if `is_complete` is a Truthy value because it is some datetime object, then we could set is_complete` to True.

Ideally you'd write a helper method and you could call it here.